torch/csrc/autograd/python_anomaly_mode.cpp

```diff
@@ -30,12 +32,55 @@ void PyAnomalyMetadata::store_stack() {
 
 void PyAnomalyMetadata::print_stack(const std::string& current_node_name) {
   pybind11::gil_scoped_acquire gil;
-  if (!PyDict_Check(dict())) {
+  _print_stack(dict(), ANOMALY_TRACE_KEY, current_node_name, false);
+  PyObject* pyparent(PyDict_GetItemString(dict(), ANOMALY_PARENT_KEY));
+
+  // if there is no "parent_" in metadata, then it means this metadata's node
+  // is the root and stop printing the traceback
+  while (pyparent) {
+    PyObject* parent_metadata(PyObject_GetAttrString(pyparent, "metadata"));
+    if (!parent_metadata) {
+      throw python_error();
+    }
+    PyObject* parent_name_pyobj(PyObject_CallMethod(pyparent, "name", ""));
```

Comment:
You can access this with `pyparent->cdata.lock()->name()` I think, if you don't want to add a brand new function to `python_function.cpp`.

Reply:
It gives me this error:

    ../torch/csrc/autograd/python_anomaly_mode.cpp: In member function ‘virtual void torch::autograd::PyAnomalyMetadata::print_stack(const string&)’:
    ../torch/csrc/autograd/python_anomaly_mode.cpp:49:45: error: ‘using PyObject = struct _object {aka struct _object}’ has no member named ‘cdata’
         PyObject* parent_name_pyobj = pyparent->cdata.lock()->name();

If there is no harm in creating a `name()` function in `python_function.cpp`, I will leave it like this.

Reply:
Ho right you would need to cast it to the proper type before taking the cdata. But you can have a few types here. The name function is fine here.

Reply:
FYI, depending on where the Node comes from, this will call either your new function or [this](https://github.com/pytorch/pytorch/blob/73dcfc5e786258dd5beeff9c313d31213640a1de/torch/csrc/autograd/python_cpp_function.cpp#L156-L159) one ;)